### DISCUSSION TITLE
Query about depth functionality in nlohmann::json

### DISCUSSION THREAD

**Discussion by 442701102** [nlohmann/json#4125] [Q&A]
"Hello, I've been using the nlohmann::json library. I searched for the keyword depth in the source code but couldn't find relevant information. I noticed that there doesn't seem to be a direct way to get the depth of a JSON object. Specifically, I was looking into the json_sax_dom_parser class and didn't find an interface to retrieve the depth. Is there a built-in method to achieve this? If not, may I know the design rationale behind not including this feature? I believe having a method to get the depth could be beneficial for many use cases. Thanks for any insights!"

**Comment by gregmarr** [comment]
> If not, may I know the design rationale behind not including this feature?

I would say that it's simply that no one has ever asked for it.  I don't recall seeing any requests for it.

**Comment by nlohmann** [comment]
A JSON value has no information where it is located in the hierarchy. That said, to know the depth, one would need to traverse from a root element to be given by the user. There is a parent relation of the `JSON_DIAGNOSTICS` macro is set, but then again, nobody so far was interested in this information.
